design-guidelines/form-ux.md

```diff
@@ -0,0 +1,189 @@
+
+### Required field indicator
+
+When there is a collection of optional and required fields, the required fields should be clearly marked. This can help the user avoid submitting an invalid form because they left a required field blank.
+
+![Input field diagram - required indicator](https://raw.githubusercontent.com/dcos-labs/ui-kit/master/design-guidelines/img/form-ux-patterns/inputFieldDiagram-requiredIndicator.png)
+
+### Label tooltip
+
+Describes the purpose of the field and how it's value is used to help the user complete their task. This information will be hidden until the user opens the Tooltip, so do not put any information in here that is critical to task completion.
+
+![Input field diagram - tooltip](https://raw.githubusercontent.com/dcos-labs/ui-kit/master/design-guidelines/img/form-ux-patterns/inputFieldDiagram-tooltip.png)
+
+### Input
+
+The input is the component that users interact with to set the value of the field.
+
+![Input field diagram - input](https://raw.githubusercontent.com/dcos-labs/ui-kit/master/design-guidelines/img/form-ux-patterns/inputFieldDiagram-input.png)
+
+### Placeholder text
+
+When an input does not have a value set, placeholder text may be shown in it's place. Placeholder text is useful for hinting what data the input is expecting to receive. For example: showing "1" in the placeholder text for a field that is expecting a number.
+
+Labels may not be replaced by placeholder text unless there are enough visual cues or context clues for the user to understand what the input does. Refer to the "Label" section above for information on labeling inputs.
+
+![Input field diagram - placeholder](https://raw.githubusercontent.com/dcos-labs/ui-kit/master/design-guidelines/img/form-ux-patterns/inputFieldDiagram-placeholder.png)
+
+### Hint content
+
+Hint text can be used to caption an input with information that helps the user fill in the correct data. Hint text can also be used to display validation errors specific to the field.
+
+Hint content should be kept as short as possible.
+
+![Input field diagram - hint content](https://raw.githubusercontent.com/dcos-labs/ui-kit/master/design-guidelines/img/form-ux-patterns/)
+
+## Input states
+
+See the [interactive component states guidelines](https://raw.githubusercontent.com/dcos-labs/ui-kit/master/design-guidelines/interactive-element-states.md) for more infomation about the various states an input can be in.
+
+## Submitting a form
+
+Forms can be submitted using the "Enter" key, or a submit button.
+
+The form's submit button should be highly visible, and clearly labeled to communicate what will happen when the button is pressed. For example: a button that uses form data to create a new object could have a label that says "Create", and a button that uses form data to update an existing object could have a label that says "Update" or "Save". forms
```

Comment:
dangling ` forms` at the end of line.

don't know why i can't make text-suggestions in a review... 🤔